pkg/calculator/handler.go

```diff
@@ -16,15 +18,63 @@ func NewEmissionCalculator(eventBus bus.Bus) *EmissionCalculator {
 	}
 }
 
-func (c *EmissionCalculator) Apply(event bus.Event) {
+func (ec *EmissionCalculator) Apply(event bus.Event) {
 	// Make sure we got the right event
 	if metricsCollected, ok := event.(v1.MetricsCollected); ok {
```

Comment:
Just a nitpick but we should stick to return often/return early here else we have a huge amount of logic in an if statement where this could be:
```go
metricsCollected, ok := event.(v1.MetricsCollected)

// handle unknown events
if !ok {
	klog.Errorf("EmissionCalculator got an unknown event: %+v", event)
	return
}

// logic follows here

```

Reply:
Really good point!

Reply:
There are some additional aspects to consider:
- One handler might handle multiple events, so the above might not always be applicable
- We can have a dedicated function which just accepts the `v1.MetricsCollected` and returns an event, so inside the conditional block there is only the call to that function